Chap_API_Sharing_Basics.tex

```diff
@@ -0,0 +1,630 @@
+%% The pmix_info_t type (See \ref{chap:introduction:portability:attributes}), 
+%% is also built on key-values pairs.  
+%% However, pmix_info_t is used for influencing API calls and is not used for sharing 
+%% information between processes.  
+This chapter describes how key-value pairs are made available and accessed by clients.
+
+A key-value pair has a scope defined by its data realm.  
+PMIx defines data realms for sessions, jobs, applications, nodes, processes and a special 
+realm for publish/lookup information.   
+These are explained more fully in chapters \ref{chap:api_rsvd_keys} and \ref{chap:pub}.
+
+The source or provider of key-value pairs can be the host environment or a \ac{PMIx} client.  
+The host environment can provide key-values pairs in any realm except for the publish/lookup realm.
+%%% Can we have a separate effort to change this restriction on publish/lookup for consistency?
+%%% Would this get into too many other issues?
+PMIx clients can only provide key-values pairs within their own process realm and the publish/lookup realm.
+The value of a key is not necessarily static and can be modified by the original provider of the key.
+
+\ac{PMIx} classifies keys as either reserved or non-reserved.
+
+Only keys provided by a host environment or \ac{PMIx} implementation can begin with \code{"pmix"} and are called reserved keys.  
+Host environments and \ac{PMIx} implementations must only provide keys that begin with \code{"pmix"} to avoid conflicts with keys that are provided by clients.
+Reserved keys are documented in chapter \ref{chap:api_rsvd_keys}.
+Each standardized reserved key has an associated macro that is documented along with its description.
+Likewise clients can only provide keys that do not begin with \code{"pmix"} to avoid conflicting with keys provided by the host environment or \ac{PMIx} implementation.   
+The string representation and the macro are equivalent methods to refer to a key.
+Throughout this document, the macro is used rather than its string value to refer to reserved keys.
+Clients cannot create reserved keys, nor can they modify the value of reserved keys.
+The way reserved keys are accessed by clients 
+is different than for non-reserved keys as explained in \ref{chap:api_rsvd_keys}.
+
+Host environments are not required to provide all keys that are documented in the standard.  
+However, when providing keys that are documented, they must adhere to the behavior defined by the standard for reserved keys.  
+An implementation cannot provide a reserved key that behaves differently than what is in the standard.  
+Implementations are encouraged to standardize any non-standardized, reserved keys that they provide to avoid conflicting with other implementations or efforts to standardize the same key.
+
+Host environments are permitted to provide non-reserved keys and non-standardized reserved keys.  
+However there are important caveats to providing either of these categories of keys 
```

Comment:
Err - I don't think this is a good idea. High potential for conflicts between the host and implementation on reserved keys. Hosts can certainly provide non-reserved keys if they choose - advise them to prefix accordingly (e.g., `LSF_FOO` and `lsf.foo`).